apps/web/middleware.ts

```diff
@@ -8,6 +9,22 @@ import { extendEventData, nextCollectBasicSettings } from "@calcom/lib/telemetry
 const middleware: NextMiddleware = async (req) => {
   const url = req.nextUrl;
 
+  if (!url.pathname.startsWith("/api")) {
+    try {
+      // Check whether the maintenance page should be shown
+      const isInMaintenanceMode = await get<boolean>("isInMaintenanceMode");
```

Comment:
Be aware that this middleware doesn't run on all routes. It only runs if the path is matched in the config on line 70. That config was added now so long ago, because before that the middleware was indeed ran on all of the routes, which was quite costly as far as I remember.

Reply:
I tried earlier on to have it without that config, but that also meant this logic was ran on non html pages like _next - brought back that config.

The reason this works although effectively the only page is /auth/login is because the user is redirected to the login page on trpc error (I think it has an unauth handler that triggers on error).